client/main/tableview.js

```diff
@@ -761,16 +767,82 @@ const TableView = SilkyView.extend({
-                                type: 'error'
-                            };
+                        else if ( ! this.currentColumn.autoMeasure) {
+                            let found = false;
+
+                            if (typeof value === 'string') {
+                                for (let levelInfo of this.currentColumn.levels) {
+                                    if (value === levelInfo.label) {
+                                        value = levelInfo.value;
+                                        found = true;
+                                        break;
+                                    }
+                                }
+                            }
+
+                            if ( ! found) {
+
+                                return new Promise((resolve, reject) => {
+                                    keyboardJS.pause();
+                                    this._editing = false;
+                                    setTimeout(() => {
+                                        dialogs.prompt('Enter an integer value for the label \'' + value + '\'', '', (result) => {
+                                            keyboardJS.resume();
+                                            if (result === undefined)
+                                                reject({ title: 'Value', message: 'Editing cancelled.'});
+                                            let n = parseInt(result);
+                                            if (isNaN(n) || n <= 0)
+                                                reject({ title: 'Value', message: 'Nominal/Ordinal values must be a positive integer.'});
+                                            else
+                                                resolve(n);
+                                        });
+                                    }, 0);
+                                }).then(n => {
+
+                                    let newLevels = [];
+                                    let found = false;
+                                    for (let level of this.currentColumn.levels) {
+                                        if (n === level.value) {
+                                            found = level.label;
+                                            level = { value: n, label: value, importValue: level.importValue };
+                                        }
+                                        newLevels.push(level);
+                                    }
+                                    if ( ! found)
+                                        newLevels.push({ value: n, label: value, importValue: n.toString() });
+
+                                    return new Promise((resolve, reject) => {
+                                        if (found) {
+                                            keyboardJS.pause();
+                                            dialogs.confirm('The value ' + n + ' currently has the label \'' + found + '\'. Change?', (result) => {
```

Comment:
we liked this, but probably isn't necessary if the underlying value is chosen automatically